### PR TITLE
[TDV-178] Requests modules

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: src/vue/modules/requests/asset-creation-requests/store/index.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  },
+  [types.CONCAT_ASSET_CREATION_REQUESTS] (state, assetCreationRequests) {
+    state.assetCreationRequests = state.assetCreationRequests
+      .concat(assetCreationRequests)
+  },
+}
+
+export const actions = {
+  [types.LOAD_ASSET_CREATION_REQUESTS] ({ getters }) {
+    return api().getWithSignature(`/${HORIZON_VERSION_PREFIX}/create_asset_requests`, {
```
Comment (quinn71) [inline src/vue/modules/requests/asset-creation-requests/store/index.js]: Guys made an API more consistent so we can make the client too.

if it's called `create_asset_request` on API, then we should call it the same way on clients. Same for every other request type

File: src/vue/modules/requests/asset-creation-requests/store/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+export const state = {
+  accountId: '',
+  assetCreationRequests: [],
+}
+
+export const mutations = {
+  [types.SET_ACCOUNT_ID] (state, accountId) {
+    state.accountId = accountId
+  },
+  [types.SET_ASSET_CREATION_REQUESTS] (state, assetCreationRequests) {
```
Comment (quinn71) [inline src/vue/modules/requests/asset-creation-requests/store/index.js]: same about over declarative name. we have the scoped namespace of the module. it's very obvious that by `request` we mean exactly `asset_creation` requests

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.spec.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          }),
+        })
+
+        expect(wrapper.vm.canBeUpdated).to.be.true
+      })
+
+      it('returns false for non-rejected and non-pending request', () => {
+        wrapper.setProps({
+          request: new AssetCreationRequest({
+            stateI: 0,
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.spec.js]: Fixed, used request state

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.spec.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        wrapper.setProps({
+          request: new AssetCreationRequest({ id: '1' }),
+        })
+        await wrapper.vm.cancelRequest()
+
+        expect(wrapper.vm.cancelAssetCreationRequest)
+          .to.have.been.calledOnceWithExactly('1')
+      })
+
+      it('calls Bus.success if the request was canceled', async () => {
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.spec.js]: Renamed

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-attributes-viewer.spec.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    beforeEach(() => {
+      Config.initConfig({
+        horizonURL: 'https://horizon.com',
+        storageURL: 'https://storage.com',
+      })
+    })
+
+    describe('assetTermsUrl', () => {
+      it('returns request.termsUrl', () => {
+        const expectedTermsUrl = 'https://test.com/terms'
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-attributes-viewer.spec.js]: Renamed

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+    },
+
+    canBeCanceled () {
+      return this.request.isPending
+    },
+  },
+
+  methods: {
+    ...mapActions('asset-creation-requests', {
+      cancelAssetCreationRequest: types.CANCEL_ASSET_CREATION_REQUEST,
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-actions-bar.vue]: Renamed

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-attributes-viewer.vue [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  </div>
+</template>
+
+<script>
+import { AssetCreationRequest } from '../wrappers/asset-creation-request'
+
+import { config } from '../_config'
+
+export default {
+  name: 'asset-creation-request-attributes-viewer',
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-attributes-viewer.vue]: Renamed

File: src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-viewer.vue [hunk 8]
```diff
@@ -0,0 +1,76 @@
+<template>
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/components/asset-creation-request-viewer.vue]: Renamed

File: src/vue/modules/requests/asset-creation-requests/index.spec.js [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+          wrapper.vm.loadAssetCreationRequests.restore()
+        })
+
+        it('returns the response of loadAssetCreationRequests method', async () => {
+          sinon.stub(wrapper.vm, 'loadAssetCreationRequests')
+            .resolves({ data: {} })
+
+          const result = await wrapper.vm.loadRequests()
+
+          expect(result).to.deep.equal({ data: {} })
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/index.spec.js]: Fixed

File: src/vue/modules/requests/asset-creation-requests/store/index.js (hunk 1 above)
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/store/index.js]: Renamed:
- `asset-creation-requests` -> `create-asset-requests`
- `asset-update-requests` -> `update-asset-requests`
- `sale-creation-requests` -> `create-sale-requests`

File: src/vue/modules/requests/asset-creation-requests/store/index.js (hunk 2 above)
Comment (ardier16) [reply src/vue/modules/requests/asset-creation-requests/store/index.js]: Renamed

File: src/vue/forms/AssetCreateForm.vue [hunk 10]
```diff
@@ -293,7 +291,7 @@ export default {
   },
   mixins: [FormMixin],
   props: {
-    request: { type: AssetCreateRequestRecord, default: null },
+    request: { type: Object, default: null },
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/forms/AssetCreateForm.vue]: Cause `CreateAssetRequest` record is now declared inside the specific module, not in the global scope, therefore `AssetCreateForm` has no clue about that kind of record

File: src/vue/forms/AssetUpdateForm.vue [hunk 11]
```diff
@@ -185,7 +185,7 @@ export default {
   },
   mixins: [FormMixin],
   props: {
-    request: { type: AssetUpdateRequestRecord, default: null },
+    request: { type: Object, default: null },
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/forms/AssetUpdateForm.vue]: Cause `UpdateAssetRequest` record is now declared inside the specific module, not in the global scope, therefore `AssetUpdateForm` has no clue about that kind of record

File: src/vue/forms/CreateSaleForm.vue [hunk 12]
```diff
@@ -390,8 +389,9 @@ export default {
   mixins: [FormMixin],
   props: {
     request: {
-      type: SaleRequestRecord,
-      default: _ => (new SaleRequestRecord()) },
+      type: Object,
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/forms/CreateSaleForm.vue]: Cause `CreateSaleRequest` record is now declared inside the specific module, not in the global scope, therefore `CreateSaleForm` has no clue about that kind of record

File: src/vue/forms/CreateSaleForm.vue [hunk 13]
```diff
@@ -390,8 +389,9 @@ export default {
...
+      default: _ => ({}),
```
[the remark this answers is not in the document]
Comment (ardier16) [reply src/vue/forms/CreateSaleForm.vue]: We can't, eslint doesn't allow us to do that

File: src/vue/forms/AssetUpdateForm.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (quinn71) [reply src/vue/forms/AssetUpdateForm.vue]: @ardier16 than create the wrapper in the scope of the module, so `AssetUpdateForm` will have a clue

File: CHANGELOG.md [hunk 14]
```diff
@@ -39,6 +44,7 @@ for further information about branching and tagging conventions.
 - "Fee" column from the movements table
 - Transfer & deposit pseudomodules from the loyalty points reconciliation
   scheme
+- common records wrappers
```
Comment (max-nes) [inline CHANGELOG.md]: Please list the removed wrappers

File: CHANGELOG.md [hunk 15]
```diff
@@ -171,6 +177,14 @@ for further information about branching and tagging conventions.
 #### Fixed
 - Not handling 'manage-asset-pair' operation details
 
+### Removed
+- `AssetRequestDetails` component
+- `SaleRequestDetails` component
+- `PreIssuanceCreateRequestRecord` wrapper
+- `SaleRequestRecord` wrapper
+- Update sale details JSON mock
+- Sale create request JSON mock
```
Comment (max-nes) [inline CHANGELOG.md]: these ones look like the list of "common records wrappers". anyway, they should be under "unreleased"

**Pull request merged**